glib-macros/src/gboxed_shared_derive.rs

```diff
@@ -178,5 +178,40 @@ pub fn impl_gshared_boxed(input: &syn::DeriveInput) -> proc_macro2::TokenStream
         #impl_to_value_optional
 
         #impl_from_value
+
+        impl #crate_ident::translate::FromGlibPtrNone<*const #refcounted_type_prefix::InnerType> for #name {
+            #[inline]
+            unsafe fn from_glib_none(ptr: *const #refcounted_type_prefix::InnerType) -> Self {
+                let ptr = #refcounted_type_prefix::ref_(ptr);
+                #name(#refcounted_type_prefix::from_raw(ptr))
+            }
+        }
+
+        impl #crate_ident::translate::FromGlibPtrFull<*mut #refcounted_type_prefix::InnerType> for #name {
+            #[inline]
+            unsafe fn from_glib_full(ptr: *mut #refcounted_type_prefix::InnerType) -> Self {
+                #name(#refcounted_type_prefix::from_raw(ptr))
+            }
+        }
+
+        impl<'a> #crate_ident::translate::ToGlibPtr<'a, *const #refcounted_type_prefix::InnerType> for #name {
+            type Storage = &'a Self;
+
+            #[inline]
+            fn to_glib_none(&'a self) -> #crate_ident::translate::Stash<'a, *const #refcounted_type_prefix::InnerType, Self> {
+                let r = <#name as #crate_ident::subclass::shared::SharedType>::into_refcounted(self.clone());
+                unsafe {
+                    #crate_ident::translate::Stash(#refcounted_type_prefix::into_raw(r), self)
```

Comment:
I think this one is the wrong one. It shouldn't `into_raw()` but just get a pointer to the existing one